examples/progress_bar_threading.py

```diff
@@ -0,0 +1,58 @@
+"""This file provides a minimal working example using a progress bar alongside
+@thread_worker to report progress.
+"""
+from time import sleep
+from qtpy.QtWidgets import QPushButton, QVBoxLayout, QWidget
+import napari
+from napari.qt import thread_worker
+
+viewer = napari.Viewer()
+
+def handle_yields(yielded_val):
+    print(f"Just yielded: {yielded_val}")
+
+# generator thread workers can provide progress updates on each yield
+@thread_worker(
+    # passing a progress dictionary with the total number of expected yields
+    # will place a progress bar in the activity dock and increment its value
+    # with each yield. We can optionally pass a description for the bar
+    # using the 'desc' key.
+    progress={'total': 5, 'desc':'thread-progress'},
+    # this does not preclude us from connecting other functions to any of the
+    # worker signals (including `yielded`)
+    connect={'yielded': handle_yields} 
+    )
+def my_long_running_thread(_):
+    for i in range(5):
+        sleep(0.1)
+        yield i
+
+# in the previous example, yielding values beyond the total value passed to
+# the progress dictionary simply leaves the progress bar on 100% with 
+# no further indicator.
+@thread_worker(
+    # If we are unsure of the number of expected yields,
+    # we can instruct the progress bar to become indeterminate once the 
+    # expected number of yields is exceeded using `may_exceed_total`
```

Comment:
my first inclination would have been to just do this by default (no option provided)... That is, if worker is still yielding even when it said it should be done, I'd imagine we want to show that somehow... and the indeterminate state seemed like a good way to do that.  Did you run into a scenario that made you think that should not be the default behavior?

Reply:
Yeah you're right, not becoming indeterminate is just a less useful option. Fixed that now.